pkg/security/jwt5/validator.go

```diff
@@ -0,0 +1,82 @@
+package jwt5
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v5"
+	pkgErrors "github.com/plgd-dev/hub/v2/pkg/errors"
+)
+
+type Validator struct {
+	keys *KeyCache
+}
+
+var (
+	ErrMissingToken     = errors.New("missing token")
+	ErrCannotParseToken = errors.New("could not parse token")
+)
+
+func NewValidator(keyCache *KeyCache) *Validator {
+	return &Validator{keys: keyCache}
+}
+
+func errParseTokenInvalidClaimsType(t *jwt.Token, err error) error {
+	return pkgErrors.NewError(fmt.Sprintf("unsupported type %T", t.Claims), ErrCannotParseToken, err)
+}
+
+func (v *Validator) ParseToken(token string) (*jwt.Token, error) {
+	if token == "" {
+		return nil, ErrMissingToken
+	}
+	return jwt.Parse(token, v.keys.GetOrFetchKey, jwt.WithIssuedAt())
+}
+
+func (v *Validator) Parse(token string) (jwt.MapClaims, error) {
+	t, err := v.ParseToken(token)
+	if t == nil {
```

Comment:
`err != nil`

Reply:
this seems to be a deliberate behavior, at least some tests rely that you get the token even if you get an error. Do we want to modify it?